bfdriver/src/platform/efi/src/bfloader.c

```diff
@@ -0,0 +1,77 @@
+//
+// Bareflank Hypervisor
+// Copyright (C) 2018 Assured Information Security, Inc.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#include "bfefi.h"
+#include "bflib.h"
+#include "bfloader.h"
+#include "bfmodule.h"
+#include "start.h"
+#include "common.h"
+
+EFI_HANDLE this_image_h;
+EFI_MP_SERVICES_PROTOCOL *g_mp_services;
+
+EFI_STATUS efi_main(EFI_HANDLE image_in, EFI_SYSTEM_TABLE *st_in)
+{
+
+    bf_init_lib(image_in, st_in);
+
+    Print(L"=======================================\n");
+    Print(L" ___                __ _           _   \n");
+    Print(L"| _ ) __ _ _ _ ___ / _| |__ _ _ _ | |__\n");
+    Print(L"| _ \\/ _` | '_/ -_)  _| / _` | ' \\| / /\n");
+    Print(L"|___/\\__,_|_| \\___|_| |_\\__,_|_||_|_\\_\\\n");
+    Print(L"     EFI Loader  \n");
+    Print(L"=======================================\n");
+
+    EFI_STATUS status;
+
+    EFI_GUID gEfiMpServiceProtocolGuid = EFI_MP_SERVICES_PROTOCOL_GUID;
+    status = gBS->LocateProtocol(&gEfiMpServiceProtocolGuid,
+                                 NULL,
+                                 (VOID **)&g_mp_services);
+    if (EFI_ERROR(status)) {
+        PRINT_ERROR(status);
+        goto fail;
+    }
+
+    add_hypervisor_modules();
+
+    Print(L"Loading modules..\n");
+    int64_t ret = common_load_vmm();
+    if (ret < 0) {
+        Print(L"common_load_vmm returned %a\n", ec_to_str(ret));
+        goto fail;
+    }
+
+    //bf_start_by_interactive();
```

Comment:
The commented-out calls should be removed. We could enable the different start modes by a compiler flag or some config argument